test/built-ins/Proxy/enumerate/return-trap-result-done.js

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 9.5.11
+description: >
+    Trap returns a done iterable result.
+info: >
+    [[Enumerate]] ()
+
+    11. Return trapResult
+---*/
+
+var x;
+var p = new Proxy([1,2,3], {
+    enumerate: function(t) {
+        return {next: function() { return { done:true }; } };
+    }
+});
+
+for (x in p) {
+    $ERROR("returned iterable interface from trap is flagged as done.");
```

Comment:
@evilpie and here we have a specific test based on your feedback to check the trap return as a done iterable. This might serve as a flag to check when this feature will be fixed on Firefox or any other envs.